MaskRCNN_no_batch_convergence/eval.py

```diff
@@ -158,6 +206,63 @@ def predict_dataflow(df, model_func, tqdm_bar=None):
     return all_results
 
 
+
+def predict_dataflow_batch(df, model_func, tqdm_bar=None):
+    """
+    Args:
+        df: a DataFlow which produces (image, image_id)
+        model_func: a callable from the TF model.
+            It takes image and returns (boxes, probs, labels, [masks])
+        tqdm_bar: a tqdm object to be shared among multiple evaluation instances. If None,
+            will create a new one.
+
+    Returns:
+        list of dict, in the format used by
+        `DetectionDataset.eval_or_save_inference_results`
+    """
+    df.reset_state()
+    all_results = []
+    with ExitStack() as stack:
+        # tqdm is not quite thread-safe: https://github.com/tqdm/tqdm/issues/323
+        if tqdm_bar is None:
+            tqdm_bar = stack.enter_context(get_tqdm(total=df.size()))
+        for imgs, img_ids, orig_sizes in df:
+            results = predict_image_batch(imgs, model_func, orig_sizes)
+            batch_id = 0
+            for img_results in results:
+                for r in img_results:
+
+                    img_id = int(img_ids[batch_id])
+                    class_id = int(r.class_id)
+                    bbox = list([float(b) for b in r.box])
+                    score = round(float(r.score), 4)
+
+                    print("A result")
```

Comment:
Let's comment these prints out. They are debugging statements that I hadn't gotten around to disabling.